atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameGetter.java

```diff
@@ -67,6 +77,44 @@ public String getInternalShortOverflowTableName(ConnectionSupplier connectionSup
         return getPrefixedOverflowTableName(tableRef);
     }
 
+    public Set<TableReference> getTableReferencesFromShortTableNames(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+        return getTableReferencesFromShortTableNamesWithPrefix(connectionSupplier, shortTableNames, tablePrefix);
+    }
+
+    public Set<TableReference> getTableReferencesFromShortOverflowTableNames(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+        return getTableReferencesFromShortTableNamesWithPrefix(
+                connectionSupplier, shortTableNames, overflowTablePrefix);
+    }
+
+    private Set<TableReference> getTableReferencesFromShortTableNamesWithPrefix(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames, String tablePrefixToStrip)
+            throws TableMappingNotFoundException {
+        Set<String> longTableNames = getLongTableNamesFromTableNames(connectionSupplier, shortTableNames);
+        return longTableNames.stream()
+                .peek(tableName -> {
+                    if (!StringUtils.startsWithIgnoreCase(tableName, tablePrefixToStrip)) {
```

Comment:
alternative would be to silently pass it through, but there's a bug if we're loading physical tables without the prefix!